### PR TITLE
Random fixes for logger connector PoC

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #7631

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: tests/trainer/logging_/test_train_loop_logging.py [hunk 1]
```diff
@@ -781,5 +787,19 @@ def training_step(self, *args):
 
     trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=1)
     model = TestModel()
-    with pytest.raises(ValueError, match=rf"self.log\(foo, {value}\)` was called"):
+    match = escape(f"self.log(foo, {value})` was called")
```
Comment (awaelchli) [inline tests/trainer/logging_/test_train_loop_logging.py:790]: did not know that!

**Review by awaelchli — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -33,6 +33,9 @@
 /pytorch_lightning/tuner                @SkafteNicki @borda @awaelchli
 /pytorch_lightning/utilities            @borda @tchaton @SeanNaren @carmocca
 
+# Specifics
+/pytorch_lightning/trainer/connectors/logger_connector @tchaton @carmocca
```
Comment (Borda) [inline .github/CODEOWNERS:37]: pls at leats three people

File: pytorch_lightning/callbacks/pruning.py [hunk 3]
```diff
@@ -259,13 +259,13 @@ def _create_pruning_fn(self, pruning_fn: str, **kwargs: Any) -> Union[Callable,
     def _wrap_pruning_fn(pruning_fn: Callable, **kwargs: Any) -> Callable:
         return partial(pruning_fn, **kwargs)
 
-    def make_pruning_permanent(self, pl_module: LightningModule) -> None:
+    def make_pruning_permanent(self, module: nn.Module) -> None:
```
Comment (Borda) [inline pytorch_lightning/callbacks/pruning.py:262]: back compatibility?

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/callbacks/pruning.py (hunk 3 above)
Comment (carmocca) [reply pytorch_lightning/callbacks/pruning.py:262]: It's considered beta, no need
